In [27]:
#

def digitArrange(n = 111111):
    #n is a non negative integer
    n = str(n)
    n = sorted(n, reverse = True)
    result = ''
    for i in n:
        result += i
    return int(result)
    

In [28]:
digitArrange()

111111

In [29]:
digitArrange(584366)

866543

In [32]:
import pandas as pd
import numpy as np

file_to_load = 'JEOPARDY_CSV.csv'

jeopardy_df = pd.read_csv(file_to_load)

In [33]:
jeopardy_df.head()

Show Number    Air Date      Round                         Category  Value  \
0         4680  12/31/2004  Jeopardy!                          HISTORY  $200    
1         4680  12/31/2004  Jeopardy!  ESPN's TOP 10 ALL-TIME ATHLETES  $200    
2         4680  12/31/2004  Jeopardy!      EVERYBODY TALKS ABOUT IT...  $200    
3         4680  12/31/2004  Jeopardy!                 THE COMPANY LINE  $200    
4         4680  12/31/2004  Jeopardy!              EPITAPHS & TRIBUTES  $200    

                                            Question      Answer  
0  For the last 8 years of his life, Galileo was ...  Copernicus  
1  No. 2: 1912 Olympian; football star at Carlisl...  Jim Thorpe  
2  The city of Yuma in this state has a record av...     Arizona  
3  In 1963, live on "The Art Linkletter Show", th...  McDonald's  
4  Signer of the Dec. of Indep., framer of the Co...  John Adams

In [41]:
jeopardy_df.describe()

Show Number
count  216930.000000
mean     4264.238519
std      1386.296335
min         1.000000
25%      3349.000000
50%      4490.000000
75%      5393.000000
max      6300.000000

2a. There are 216,930 rows.  Each row represents a jeopardy question, answer, and other information.

In [60]:
jeopardy_df.columns = jeopardy_df.columns.str.rstrip()
jeopardy_df.columns = jeopardy_df.columns.str.lstrip()

jeopardy_df.columns

Index(['Show Number', 'Air Date', 'Round', 'Category', 'Value', 'Question',
       'Answer', 'Year'],
      dtype='object')

In [65]:
jeopardy_df['Year'] = pd.DatetimeIndex(jeopardy_df['Air Date']).year

#print(jeopardy_df.head())

unique_shows_per_year = jeopardy_df.groupby('Year')['Show Number'].nunique()
print(unique_shows_per_year)

Year
1984     23
1985     16
1986     25
1987     22
1988     22
1989     36
1990     73
1991     24
1992     32
1993     35
1994     19
1995     19
1996     83
1997    221
1998    223
1999    229
2000    225
2001    203
2002    114
2003    158
2004    220
2005    225
2006    230
2007    231
2008    232
2009    226
2010    231
2011    224
2012     19
Name: Show Number, dtype: int64


2b. See above output.

In [72]:
values = jeopardy_df.groupby('Year')['Value']
values.head()

#I could try to strip out the '$' and convert to an int, but it's late....

0        $200 
1        $200 
2        $200 
3        $200 
4        $200 
         ...  
13726    $100 
13727    $100 
13728    $100 
13729    $100 
13730    $100 
Name: Value, Length: 145, dtype: object